test/test_pipeline/test_regression.py

```diff
@@ -435,3 +436,152 @@ def test_set_params(self):
 
     def test_get_params(self):
         pass
+
+    def _test_set_hyperparameter_choice(self, expected_key, implementation, config_dict):
+        """
+        Given a configuration in config, this procedure makes sure that
+        the given implementation, which should be a Choice component, honors
+        the type of the object, and any hyperparameter associated to it
+        """
+        keys_checked = [expected_key]
+        implementation_type = config_dict[expected_key]
+        expected_type = implementation.get_components()[implementation_type]
+        keys_checked.append(expected_key)
```

Comment:
In the end I compare set() because we are dealing with dict keys, so functionally it is not an issue but clearly doesn't look nice. I have fixed this, thank for pointing this out!